src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/PooledTxsRequestor.cs

```diff
@@ -112,15 +120,15 @@ private void AddMarkUnknownHashes(IReadOnlyList<Keccak> hashes, ArrayPoolList<Ke
                 }
             }
         }
-        private void AddMarkUnknownHashesEth68(IReadOnlyList<Keccak> hashes, IReadOnlyList<int> sizes, ArrayPoolList<(Keccak, int)> discoveredTxHashesAndSizes)
+        private void AddMarkUnknownHashesEth68(IReadOnlyList<Keccak> hashes, IReadOnlyList<int> sizes, IReadOnlyList<byte> types, ArrayPoolList<(Keccak, byte, int)> discoveredTxHashesAndSizes)
         {
             int count = hashes.Count;
             for (int i = 0; i < count; i++)
             {
                 Keccak hash = hashes[i];
-                if (!_txPool.IsKnown(hash) && _pendingHashes.Set(hash))
+                if (!_txPool.IsKnown(hash) && !_txPool.ContainsTx(hash, (TxType)types[i]) && _pendingHashes.Set(hash))
```

Comment:
`!_txPool.IsKnown(hash)` and `!_txPool.ContainsTx(hash, (TxType)types[i])` seems like the same thing. Can IsKnown cover what ContainsTx additionally covers?

Reply:
`IsKnown` is checking if tx hash is in cache:
`public bool IsKnown(Keccak? hash) => hash != null ? _hashCache.Get(hash) : false;`

`ContainsTx` is checking if tx is in TxPool/BlobPool.

BlobTxs are persistent (saved to db), cache would be empty after restart or just after some time blob hash would be replaced by newer cached hash. It is to avoid rerequesting txs already present in the pool.

`IsKnown` is used in other places (`TxBundleSimulator` or in `RemoveProcessedTransactions` in `TxPool` to prepare metrics), so I didn't want to modify it